packages/react/src/date-picker/DatePicker.test.tsx

```diff
@@ -24,8 +24,48 @@ jest.mock('use-debounce', () => ({
 	}),
 }));
 
-function renderDatePicker(props: DatePickerProps) {
-	return render(<DatePicker {...props} />);
+// Mock the current date so snapshot outputs are always consistent
+const mockSystemTime = new Date(2015, 7, 5);
+beforeAll(() => {
+	jest.useFakeTimers({ advanceTimers: true });
+	jest.setSystemTime(mockSystemTime);
+});
+
+afterAll(() => {
+	jest.useRealTimers();
+});
+
+type ControlledDatePickerProps = Omit<DatePickerProps, 'value' | 'onChange'> & {
+	initialValue?: DatePickerProps['value']; // value is not allowed as it is controlled, but a `initialValue` can be passed in
+	onChange?: DatePickerProps['onChange']; // onChange is optional
+};
+
+function renderDatePicker(props: ControlledDatePickerProps) {
+	return render(<ControlledDatePicker {...props} />);
+}
+
+function ControlledDatePicker({
+	onChange: onChangeProp,
+	initialValue: initialValueProp,
```

Comment:
To match the type above
```suggestion
	initialValue: initialValueProp,
	onChange: onChangeProp,
```